### PR TITLE
Bug 1872080: Fix Dockerfile

### PR DESCRIPTION
**Pull request by soltysh**
/assign @tnozicka

### CODE REVIEW

**Comment by crawford** [comment]
From the [logs](https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/openshift_cluster-openshift-apiserver-operator/385/pull-ci-openshift-cluster-openshift-apiserver-operator-master-images/1298929618379608064):
> could not run steps: step cluster-openshift-apiserver-operator failed: could not wait for build: the build cluster-openshift-apiserver-operator failed after 34s with reason ManageDockerfileFailed: Failed to prepare the dockerfile for the build

/retest

**Comment by crawford** [comment]
> error: open /tmp/build/inputs/Dockerfile: no such file or directory

This PR removes ./Dockerfile, so that build failure makes sense.

/hold

**Comment by soltysh** [comment]
> This PR removes ./Dockerfile, so that build failure makes sense.

Right, but that was fixed in https://github.com/openshift/release/pull/11230

/retest
/hold cancel

**Comment by soltysh** [comment]
Looks like it got changed by bot, new PR in https://github.com/openshift/release/pull/11525

File: Dockerfile.rhel7 [hunk 1]
```diff
@@ -1,9 +1,10 @@
-FROM registry.svc.ci.openshift.org/ocp/builder:golang-1.14 AS builder
+FROM registry.svc.ci.openshift.org/ocp/builder:rhel-8-golang-openshift-4.6 AS builder
```
Comment (tnozicka) [inline Dockerfile.rhel7:1]: I think the direction on golang versioning is still open

File: Dockerfile.rhel7 [hunk 2]
```diff
@@ -1,9 +1,10 @@
...
 WORKDIR /go/src/github.com/openshift/cluster-openshift-apiserver-operator
 COPY . .
-RUN go build ./cmd/cluster-openshift-apiserver-operator
+ENV GO_PACKAGE github.com/openshift/cluster-kube-apiserver-operator
```
Comment (tnozicka) [inline Dockerfile.rhel7:4]: this should be autodetected

**Review by tnozicka — COMMENTED** [review]
/lgtm
modulo the builder tag, that can be reverted with other PRs if the discussion goes the other way

**Pull request closed without merging**